presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java

```diff
@@ -376,6 +378,17 @@ public static TransactionManager createTransactionManager(
         return InMemoryTransactionManager.create(config, idleCheckExecutor, catalogManager, finishingExecutor);
     }
 
+    @Provides
+    @Singleton
+    @ForWatchdog
+    public static ScheduledExecutorService createWatchdogExecutor()
+    {
+        ScheduledThreadPoolExecutor executor =
+                new ScheduledThreadPoolExecutor(1, daemonThreadsNamed("thread-watchdog"));
```

Comment:
- put into the same line
- if we only have one thread, would that cause starvation if some bad thread blocks the legit ones?

Reply:
> if we only have one thread, would that cause starvation if some bad thread blocks the legit ones?

`TimeoutThread` only executes `Thread#interrupt` and finishes. It shouldn't get blocked. I believe it's safer than spawning large amount of threads by mistake.